disco/bot/bot.py

```diff
@@ -326,11 +346,18 @@ def get_commands_for_message(self, require_mention, mention_rules, prefix, msg):
                     content = content.replace('<@{}>'.format(role), '', 1)
 
             content = content.lstrip()
-
-        if prefix and not content.startswith(prefix):
-            return []
         else:
-            content = content[len(prefix):]
+            # Scan through the prefixes to find the first one that matches.
+            # This may lead to unexpected results, but said unexpectedness
+            # should be easy to avoid. An example of the unexpected results
+            # that may occur would be if one prefix was `!` and one was `!a`.
+            if not any([content.startswith(prefix) for prefix in prefixes]):
```

Comment:
This code shouldn't check over the message content twice, just do the following;

```python
for prefix in prefixes:
    if content.startswith(prefix):
        content = content[len(prefix):]
        break
else:
    return []
```

It may also make sense to give the end user access to the prefix that was matched in the command object somewhere.

Reply:
The thing is, that doesn't work. I would use an `else` after a `for` if I could though. The whole point of the overarching if statement is to tell if any prefixes will match, instead of having to use another variable or something to check if the message actually starts with a prefix.

Reply:
Why doesn't it work?

Reply:
Oh wait. Oops. Today I learned...